src/snowflake/snowpark/_internal/packaging_utils.py

```diff
@@ -0,0 +1,251 @@
+        return inverted_dictionary
+
+    native_libraries = set()
+    native_extensions = (
+        {".pyd", ".pxd", ".dll"} if platform.system() == "Windows" else {".so"}
+    )
+    for native_extension in native_extensions:
+        glob_output = glob.glob(os.path.join(target, "**", f"*{native_extension}"))
+        if glob_output:
+            folder_to_package_map = invert_package_map(downloaded_packages_dict)
+            for path in glob_output:
+                relative_path = os.path.relpath(path, target)
+                root_directory = os.path.split(relative_path)[0]
+                if root_directory in folder_to_package_map:
+                    library_set = folder_to_package_map[root_directory]
+                    for library in library_set:
+                        if library not in native_libraries:
+                            _logger.info(f"Potential native library: {library}")
+                            native_libraries.add(library)
+    return native_libraries
+
+
+def zip_directory_contents(directory_path: str, output_path: str) -> None:
+    directory_path = Path(directory_path)
+    output_path = Path(output_path)
+    with zipfile.ZipFile(
+        output_path, "w", zipfile.ZIP_DEFLATED, allowZip64=True
+    ) as zipf:
+        for file in directory_path.rglob("*"):
+            zipf.write(file, file.relative_to(directory_path))
+
+        parent_directory = directory_path.parent
+
+        for file in parent_directory.iterdir():
+            if (
+                file.is_file()
+                and not file.match(".*")
+                and file != output_path
+                and file != directory_path
+            ):
+                zipf.write(file, file.relative_to(parent_directory))
+
+
+def add_snowpark_package(
+    result_dict: Dict[str, str], valid_packages: Dict[str, List[str]]
+) -> None:
+    if SNOWPARK_PACKAGE_NAME not in result_dict:
+        result_dict[SNOWPARK_PACKAGE_NAME] = SNOWPARK_PACKAGE_NAME
+        try:
+            package_client_version = pkg_resources.get_distribution(
+                SNOWPARK_PACKAGE_NAME
+            ).version
+            if package_client_version in valid_packages[SNOWPARK_PACKAGE_NAME]:
+                result_dict[
+                    SNOWPARK_PACKAGE_NAME
+                ] = f"{SNOWPARK_PACKAGE_NAME}=={package_client_version}"
+            else:
+                _logger.warning(
+                    f"The version of package '{SNOWPARK_PACKAGE_NAME}' in the local environment is "
+                    f"{package_client_version}, which is not available in Snowflake. Your UDF might not work when "
+                    f"the package version is different between the server and your local environment."
+                )
+        except pkg_resources.DistributionNotFound:
+            _logger.warning(
+                f"Package '{SNOWPARK_PACKAGE_NAME}' is not installed in the local environment. "
```

Comment:
When will this case happen?

Reply:
Unable to think of any case where this could happen but if it does happen due to an off-beat failure (let's say a bug in pkg_resources), it should not affect the user.